### PR TITLE
feat(logs): add metric methods for log group IncomingLogEvents and IncomingBytes

### LINKED ISSUE #30034: (aws-logs): Add metric methods for log group IncomingLogEvents and IncomingBytes
### Describe the feature

Add metric methods for log group IncomingLogEvents and IncomingBytes, similar to existing metric methods like metricInvocations on Lambda functions.

https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CloudWatch-Logs-Monitoring-CloudWatch-Metrics.html

### Use Case

Create alarms on log ingestion to ensure it is working.

### Proposed Solution

Create `metricIncomingLogEvents` and `metricIncomingBytes` and maybe a generic `metric` methods on the LogGroup class similar to other resources with metric* methods.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.139.1

### Environment details (OS name and version, etc.)

Amazon Linux 2

### PR DESCRIPTION
**Pull request by michelle-wangg**
### Issue

Closes #30034

### Reason for this change
- Create alarms on log ingestion to ensure it is working

### Description of changes

- Add metric methods for log group IncomingLogEvents and IncomingBytes, similar to existing metric methods like metricInvocations on Lambda functions.

### Description of how you validated changes
- validated with new unit tests and a new integration test

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts [hunk 1]
```diff
@@ -84,6 +84,28 @@ export interface ILogGroup extends iam.IResourceWithPolicy {
...
    */
   logGroupPhysicalName(): string;
+
+  // /**
+  //  * Return the given named metric for this Log Group
+  //  *
+  //  * @param metricName The name of the metric
+  //  * @param props Properties for the metric
+  //  */
+  // metric(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric;
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-logs/lib/log-group.ts:94]: Remove this?

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts [hunk 2]
```diff
@@ -84,6 +84,28 @@ export interface ILogGroup extends iam.IResourceWithPolicy {
...
+
+  /**
+   * Metric for the number of logs ingested
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-logs/lib/log-group.ts:97]: Maybe use the description from https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CloudWatch-Logs-Monitoring-CloudWatch-Metrics.html

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts [hunk 3]
```diff
@@ -84,6 +84,28 @@ export interface ILogGroup extends iam.IResourceWithPolicy {
...
+   *
+   * @param props Properties for the metric
+   */
+  metricIncomingLogEvents(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for the number of bytes stored
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-logs/lib/log-group.ts:104]: same as above

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts [hunk 4]
```diff
@@ -245,6 +267,64 @@ abstract class LogGroupBase extends Resource implements ILogGroup {
...
+   *   threshold: 5000000,  // 5 MB
+   *   evaluationPeriods: 1,
+   * });
+   * ```
+  */
+  public metricIncomingBytes(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('IncomingBytes', props);
+  }
+
+  private metric(_metricName: string, _props?: cloudwatch.MetricOptions): cloudwatch.Metric {
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-logs/lib/log-group.ts:321]: There are a lot more metrics supported by CloudWatch log group, so maybe make this as a public interface?

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts [hunk 5]
```diff
@@ -245,6 +267,64 @@ abstract class LogGroupBase extends Resource implements ILogGroup {
...
+   * const incomingEventsMetric = logGroup.metricIncomingLogEvents();
+   * new cloudwatch.Alarm(this, 'HighLogVolumeAlarm', {
+   *   metric: incomingEventsMetric,
+   *   threshold: 1000,
+   *   evaluationPeriods: 1,
+   * });
+   * ```
+   */
+  public metricIncomingLogEvents(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('IncomingLogs', props);
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-logs/lib/log-group.ts:293]: According to the documentation, the only valid statistic is `Sum` so we should set the statistic to sum.

**Review by GavinZZ — COMMENTED** [review]
Nice work. Left some comments.

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts (hunk 5 above)
Comment (michelle-wangg) [reply packages/aws-cdk-lib/aws-logs/lib/log-group.ts:295]: Should have addressed all the comments above.

File: packages/aws-cdk-lib/aws-logs/lib/log-group.ts [hunk 6]
```diff
@@ -245,6 +271,78 @@ abstract class LogGroupBase extends Resource implements ILogGroup {
...
+ * It sets the namespace to 'AWS/Logs' and the statistic to 'Sum' by default.
+ *
+ * The created metric is automatically associated with this log group using the `attachTo` method.
+ *
+ * Common metric names for log groups include:
+ * - 'IncomingBytes': The volume of log data in bytes ingested into the log group.
+ * - 'IncomingLogEvents': The number of log events ingested into the log group.
+ * ```
+ */
+  public metric(_metricName: string, _props?: cloudwatch.MetricOptions): cloudwatch.Metric {
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-logs/lib/log-group.ts:338]:
```suggestion
  public metric(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
```

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**